crates/core_simd/src/vector.rs

```diff
@@ -135,22 +135,26 @@ where
     /// assert_eq!(v.as_array(), &[0, 1, 2, 3]);
     /// ```
     pub const fn as_array(&self) -> &[T; LANES] {
-        &self.0
+        // SAFETY: `[T; LANES]` and `Simd<T, LANES>` have the same layout.
```

Comment:
Since layout includes alignment, I'm not sure this is quite accurate

Reply:
Would something like "`Simd<T, LANES>` can always be transmuted to `[T; LANES]`" be fine with you as the safety comment?

Reply:
Yep, or maybe something like "`Simd<T, LANES>` is an over-aligned array"